### ISSUE TITLE
Clarification for the concept of default value in options?

### ISSUE THREAD

**Issue by posiczko** [postmodern/command_kit.rb#49]
Hello!

Thanks for yet another wonderful tool! I have a quick question regarding default value in options:

I would expect that defining option like so

```ruby
      option :count, short: '-c',
                     value: {
                       type: Integer,
                       default: 1
                     },
                     desc: "Number of times"
```

would actually set `options[:count]` to `1` if no option `-c` or `--count` is defined at run time. That does not seem to be the case. Is that intentional?

Thank you!

**Comment by postmodern** [comment]
Another option would be to add a second `default:` but outside of `value:`, to always set a default value for the option:

    option :count, short: '-c',
                           value: {type: Integer},
                           default: 1

Although, not sure if having a default value which only applies when the option flag is given, but no value is given, is still useful or confusing?

sets the `:option`'s value:

    cmd --option foo

defaults to the `:option`'s `value: {default: ...}` value:

    cmd --option --other-option

defaults to the `:option`'s `default: ...` value:

    cmd --other-option

**Comment by postmodern** [comment]
@posiczko I will probably go with your suggestion and always default to the option's default value, as that's the most obvious thing to do.

However, there's an edge-case scenario I need to decide on. What should happen if an option's value isn't required and can be omitted (i.e. `--opt` and `--opt VALUE` are both valid), but also has a default? Should `--opt` set `options[:opt]` to `nil`, or should `options[:opt]` remain the default value of the option?

**Comment by postmodern** [comment]
I have made the requested changes. Not that options who's value is not required (`value: {required: false, default: ...}`) can override the default value with `nil`, if the option flag is given but there's no value following it (ex: `--flag`); this at least helps indicate that the option flag was seen but no value accompanied it. The specs I added to `spec/options_spec.rb` help clarify all of the edge-cases: https://github.com/postmodern/command_kit.rb/commit/c329592c7794a950d4272d39846141f7abcfaa44#diff-2c488910940374fe3a306cf1ffd792e69481230f25fe97cf61fd56f5f2deb6d0R136-R370

**Comment by postmodern** [comment]
This should be fixed in 0.2.1 which has just been released.
